src/store/sparql.py

```diff
@@ -13,10 +17,31 @@ def setup(self):
     def run_sparql(self, query) -> QueryResult:
         """ Runs and returns the results from a sparql query"""
         self.sparql.setQuery(query)
-        self.sparql.setReturnFormat(JSON)
+        self.sparql.setReturnFormat("json")
         return self.sparql.query()
     
-    def get_datasets(self):
+    def map_query_response_to_json(self, list_of_data):
+        nicer_list = []
+        for item in list_of_data:
+            n = {
+                "title": get_value_from_dict(item, "name"),
```

Comment:
so this only says "name" because we specified ?name in the SPARQL query. It's probably worth specifying ?title in the SPARQL query instead so the mapping is more intuitive. i.e this could then be `"title": get_value_from_dict(item, "title"),`

Reply:
it in the dataset that we are suing the title is stored under name so that is how it is accessed

Reply:
you dont need optional, in the sparql query the "?name" part is variable assignment. its us deciding we want to call the variable being returned ?name - you can just use ?title instead is all i mean.

as to why bother - longer term we want a future where data engineers or scientists provide the queries and us devs just use the results - so we want as much of the mapping of thing called A to thing called B in the query as possible, keeps the python code more inuitive (and tbh they're better placed to decide those mappings than us - areas of expertise n all).